streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java

```diff
@@ -186,6 +203,68 @@ public boolean setFlushListener(final CacheFlushListener<K, V> listener,
         return false;
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public <R> QueryResult<R> query(final Query<R> query,
+        final PositionBound positionBound,
+        final boolean collectExecutionInfo) {
+
+        final long start = System.nanoTime();
+        final QueryResult<R> result;
+
+        final QueryHandler handler = queryHandlers.get(query.getClass());
+        if (handler == null) {
+            result = wrapped().query(query, positionBound, collectExecutionInfo);
+            if (collectExecutionInfo) {
```

Comment:
Does it make sense to add execution info for this case? We pushed the query down, but did not handle it and thus the inner store would track (and we should not track)?

Reply:
Yeah, the idea was to actually be able to see everything that happened during query execution, specifically to demystify what's going on when you're debugging.

Reply:
We can do this, but should we better say "passed down", not "handled" in the message?

Reply:
I'm getting the impression that you're not a huge fan of the phrasing of these messages. :) Can we tackle this question in a follow-on fashion?

Reply:
Sure.